harness/determined/_core/_distributed.py

```diff
@@ -256,11 +256,13 @@ def _zmq_allgather(self, stuff: Any) -> List:
             return [stuff]
         logging.debug(f"Worker {self.get_rank()} beginning zmq allgather.")
         if self._is_chief:
-            worker_stuff, _ = self._chief_zmq.gather_with_polling(lambda: None)
+            worker_stuff_ranked, _ = self._chief_zmq.gather_with_polling(lambda: None)
```

Comment:
This looks good, but perhaps we should either:
* apply this rank-sort to all variations of gather/allgather in this file, or
* apply the rank-sort at a different layer

Waht do you think?

Reply:
I don't see another layer where the rank-sort would be both meaningful and possible so I will go with the first option. (rank is not available outside `DistributedContext`, and we need both workers and the chief to be aware of rank being packaged inside the message).